### ISSUE TITLE
Unsigned Int displayed as Int

### ISSUE THREAD

**Issue by lbilli** [JuliaData/DataFrames.jl#2677]
I'm not sure if it is by design, but I was slightly surprised by the way `UInt` are displayed when in a `DataFrame` compared to regular vectors, _i.e._ decimal vs. hexadecimal notation:
```julia
julia> using DataFrames

julia> v = UInt8[1,125,253]
3-element Vector{UInt8}:
 0x01
 0x7d
 0xfd

julia> DataFrame(:v => v)
3×1 DataFrame
 Row │ v     
     │ UInt8 
─────┼───────
   1 │     1
   2 │   125
   3 │   253
```

**Comment by bkamins** [comment]
Yes - we have more such inconsistencies - especially for `Bool` type. The question is - do you think it is problematic? Note that we display `eltype` on top.

CC @ronisbr

**Comment by lbilli** [comment]
Certainly not a big deal and mostly about aesthetics, yet it took me a double take to the header to make sure no unexpected conversion had happened behind the scenes.

Also, I'd argue that when dealing with bit patterns or masks, a lot is lost with decimal notation:
```julia
julia> v = [0x00ff, 0xff00]
2-element Vector{UInt16}:
 0x00ff
 0xff00

julia> DataFrame(:v => v)
2×1 DataFrame
 Row │ v      
     │ UInt16 
─────┼────────
   1 │    255
   2 │  65280
```

**Comment by bkamins** [comment]
@ronisbr - do you remember why we decided to go this way (apart from handling `Bool` as a special case, which I think we can keep as is - i.e. printing `true` and `false`)?

**Comment by ronisbr** [comment]
I think we are not handling any special cases. After a very long discussion with me, you, and @nalimilan, we decided to be consistent with `print`. What we have is exactly what is obtained from `print`, in all cases.

**Comment by ronisbr** [comment]
I forgot to mention something! If you want something close to what Julia uses by default in REPL, you can change the renderer to `show` using:

```julia
julia> show(df, renderer = :show)
3×1 DataFrame
 Row │ v
     │ UInt8
─────┼───────
   1 │  0x01
   2 │  0x7d
   3 │  0xfd
```

**Comment by bkamins** [comment]
@nalimilan - I guess, especially given the last comment by @ronisbr we can close this. OK?

**Comment by nalimilan** [comment]
Well, yeah, at least it works as intended. IIRC I advocated using `show` for most types except a few special common types (like strings) during the long discussion we had when moving to PrettyTables. Anyway we can change this after 1.0 if we want.

**Comment by lbilli** [comment]
Instead of hardcoding either one, I was wondering if it has been considered to put some of these settings in some global variable that users can easily customize to their taste (similar to how `options()` works in R).

Beside `renderer`, users might have their preferences also regarding `nosubheader`,  `show_row_number`, `hlines` to name a few.

I guess the advanced user can already achieve this by carefully overriding `Base.show` but it sounds rather cumbersome.

**Comment by bkamins** [comment]
I would prefer to have it documented how to do it. We tried very hard for years to avoid global state of DataFrames.jl, as having such a state is error prone and does not play well with multi-threading.

**Comment by ronisbr** [comment]
The very first PrettyTables.jl implementation I tried to do here in DataFrames.jl had a global state that you could modify. However, this added a huge performance loss in time to print the first table (it was almost 3x slower).
